### DISCUSSION TITLE
Proposal for new  Maven-related regexp manager preset

### DISCUSSION THREAD

**Discussion by rombert** [renovatebot/renovate#21495] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Sometimes Maven pom.xml files use properties to define versions. There are scenarios where these versions aren't directly reflected in the dependencies or dependencyManagement sections, for instance:

- docker-related plugins consume the property to set a container image label
- other plugins consume a property to set a version property in a configuration

Examples

```xml
            <plugin>
                <groupId>io.fabric8</groupId>
                <artifactId>docker-maven-plugin</artifactId>
                <version>0.42.1</version>
                <executions>
                    <execution>
                        <id>start-mongo</id>
                        <phase>pre-integration-test</phase>
                        <goals>
                            <goal>start</goal>
                        </goals>
                        <configuration>
                            <images>
                                <image>
                                    <alias>mongo</alias>
                                    <name>mongo:${mongo.container.version}</name>
                               </image>
                            </images>
                        </configuration>
                    </execution>
```

It would be beneficial if renovate would ship with a built-in present that allows users to annotate pom.xml properties, indicating that they are dependecies to be managed.

I worked on the regular expression at https://github.com/apache/sling-org-apache-sling-starter/pull/85, and I'd be willing to massage into something more generally useful, if there is intereset.

Briefly, the way to annotate the properties is

```xml
       <!-- renovate: depName=org.ow2.asm:asm -->
        <asm.version>9.3</asm.version>
```

and the regex manager is

```javascript
{
      "fileMatch": ["^pom\\.xml$"],
      "matchStrings" : [
        "depName=(?<depName>.*?)\\s+-->\\s+<.*?\\.version>(?<currentValue>.*?)<\\/.*?\\.version>"
      ],
      "datasourceTemplate": "maven"
    }
```

Does this sound like something that could be contributed to renovate?

**Comment by viceice** [comment]
sounds good.


- match full comment to be more safe
- allow more properties for the renovate comment. checkout existing regex manager presets for samples

**Reply by rombert** [reply]
Will look into it, thanks.

**Reply by rombert** [reply]
Note for my self, until I create a PR

```
<!--\s?renovate: depName=(?<depName>.*?)\s?(?: (?:lookupName|packageName)=(?<packageName>[^\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\s+-->\s+<.*?\.version>(?<currentValue>.*?)<\/.*?\.version>
```

should work fine. The datasource template should be "maven".  Code 
will go in https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/regex-managers.ts .

@viceice - would a contribution make it active by default? If not, should I do anything in particular for that?

**Reply by viceice** [reply]
yes, I think so
